fieldType.py

```diff
@@ -28,13 +28,33 @@ def __init__ (self, **kwargs):
     self.viewRank = self.rank-1
     self.viewShape = ','.join ([':'] * (self.rank-1))
     self.lbptr = ', '.join (list (map (lambda i: "LBOUNDS(" + str (i+1) + "):", range (0, self.rank))))
+    self.hasView = self.rank > 1
+    self.ganged = self.rank > 2
 
 
 kinds = ['JPRM', 'JPRB', 'JPRD', 'JPIM', 'JPLM']
 
-
-def getFieldTypeList (ranks=[2,3,4,5], kinds=kinds):
-  return [fieldType (kind=kind, rank=rank) for (kind) in kinds for rank in ranks]
+def eqv (a, b):
```

Comment:
[optional style change]

Perhaps this could be simplified a little:
```
def getFieldTypeList (ranks=[1,2,3,4,5], kinds=kinds, hasView=False, alias=True, ganged=False):

  l = [fieldType (kind=kind, rank=rank) for (kind) in kinds for rank in ranks]

  if hasView:
    l = [ft for ft in l if ft.hasView]

  if not alias:
    l = [ft for ft in l if not ft.alias]

  if ganged:
    l = [ft for ft in l if ft.ganged]
```

Reply:
isn't that just

```
l = [ft for ft in l if hasView is not None and hasView == ft.hasView]
```
or
```
if hasView is not None:
    l = [ft for ft in l if hasView == ft.hasView]
```

Reply:
Thank you Lukas, it works !

In Fortran, it is not possible to compare logicals with ==, but with python it works.